server/src/main/java/io/crate/expression/scalar/regex/RegexMatcher.java

```diff
@@ -22,70 +22,37 @@
 package io.crate.expression.scalar.regex;
 
 import javax.annotation.Nullable;
-import java.util.ArrayList;
 import java.util.List;
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 
-public class RegexMatcher {
+public abstract class RegexMatcher {
```

Comment:
kind of similar what jordi commented already. The abstract `RegexMatcher` class is implemented partially by match and replace functions, which raises the question about using this abstraction. I'd better move related to functions logic closer to the implementation and maybe add some static factory methods that would hide things like https://github.com/crate/crate/pull/10104/files#diff-bffb730cd7513f358bd585218cd15286R172 and https://github.com/crate/crate/pull/10104/files#diff-4f4a1a2d0baa1a4764044bf3363dcba8R198